File: app/src/main/java/com/studyasist/data/local/entity/ChatMessage.kt
package com.studyasist.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "chat_messages")
data class ChatMessage(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val role: String,
    val content: String,
    val subject: String? = null,
    val chapter: String? = null,
    val createdAt: Long = System.currentTimeMillis()
)

File: StudyAsist-iOS/StudyAsist/Models/ChatMessage.swift
//
//  ChatMessage.swift
//  StudyAsist
//

import Foundation
import SwiftData

@Model
final class ChatMessage {
    @Attribute(.unique) var id: Int64
    var role: String
    var content: String
    var subject: String?
    var chapter: String?
    var createdAt: Int64
    
    init(
        id: Int64 = 0,
        role: String,
        content: String,
        subject: String? = nil,
        chapter: String? = nil,
        createdAt: Int64 = Int64(Date().timeIntervalSince1970 * 1000)
    ) {
        self.id = id
        self.role = role
        self.content = content
        self.subject = subject
        self.chapter = chapter
        self.createdAt = createdAt
    }
}
